internal/databroker/server.go

```diff
@@ -145,7 +146,12 @@ func (srv *Server) Query(ctx context.Context, req *databroker.QueryRequest) (*da
 		return nil, err
 	}
 
-	_, _, stream, err := db.SyncLatest(ctx, req.GetType(), nil)
+	expr, err := storage.FilterExpressionFromStruct(req.GetFilter())
+	if err != nil {
+		return nil, err
```

Comment:
Maybe wrap an error as there are multiple error conditions in that func

Reply:
Yes good idea. I made it return `InvalidArgument` in that case.